src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -1520,6 +1520,62 @@ class D
             CompileAndVerify(compilation, expectedOutput: expectedOutput);
         }
 
+        [Fact]
+        public void ListPattern_Nullable()
+        {
+            var source = @"
+using System;
+struct S
+{
+    public int Length => 1;
+    public int this[int i] => 42;
+
+    public static bool Test(S? s)
+    {
+        return s is [42];
+    }
+
+    public static void Main()
+    {
+        Console.WriteLine(Test(new()));
+        Console.WriteLine(Test(default));
```

Comment:
nit: let's use `null` and `new S()` to be completely unambiguous

Reply:
We're not testing `new()` here though 😄 I will take your suggestion anyways.